src/DependencyInjection/MakerExtension.php

```diff
@@ -34,7 +34,29 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('services.xml');
         $loader->load('makers.xml');
 
+        $configuration = $this->getConfiguration($configs, $container);
+        $config = $this->processConfiguration($configuration, $configs);
+
+        $container->setParameter('maker.root_namespace', $config['root_namespace']);
```

Comment:
Use abstract service is great idea! I'll try it.

However where I can get `root_namespace` in DI context (compiler pass or `service.xml`) if it is not stored as "parameter"?

The second, if we are deciding to configure root namespace only, it can be more practical - to remove config and use just a parameter. What do think?

P.S.
My English leaves much to be desired. Please, fill free to correct me and ask to explain.